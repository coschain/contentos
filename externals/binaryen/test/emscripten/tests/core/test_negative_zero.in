#include <stdio.h>
#include <math.h>

int main() {
#define TEST(x, y) printf("%.2f, %.2f ==> %.2f\n", x, y, copysign(x, y));
  TEST(5.0f, 5.0f);
  TEST(5.0f, -5.0f);
  TEST(-5.0f, 5.0f);
  TEST(-5.0f, -5.0f);
  TEST(5.0f, 4.0f);
  TEST(5.0f, -4.0f);
  TEST(-5.0f, 4.0f);
  TEST(-5.0f, -4.0f);
  TEST(0.0f, 5.0f);
  TEST(0.0f, -5.0f);
  TEST(-0.0f, 5.0f);
  TEST(-0.0f, -5.0f);
  TEST(5.0f, 0.0f);
  TEST(5.0f, -0.0f);
  TEST(-5.0f, 0.0f);
  TEST(-5.0f, -0.0f);
  TEST(0.0f, 0.0f);
  TEST(0.0f, -0.0f);
  TEST(-0.0f, 0.0f);
  TEST(-0.0f, -0.0f);
  return 0;
}
