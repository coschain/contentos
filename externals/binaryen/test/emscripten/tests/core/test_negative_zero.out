5.00, 5.00 ==> 5.00
5.00, -5.00 ==> -5.00
-5.00, 5.00 ==> 5.00
-5.00, -5.00 ==> -5.00
5.00, 4.00 ==> 5.00
5.00, -4.00 ==> -5.00
-5.00, 4.00 ==> 5.00
-5.00, -4.00 ==> -5.00
0.00, 5.00 ==> 0.00
0.00, -5.00 ==> -0.00
-0.00, 5.00 ==> 0.00
-0.00, -5.00 ==> -0.00
5.00, 0.00 ==> 5.00
5.00, -0.00 ==> -5.00
-5.00, 0.00 ==> 5.00
-5.00, -0.00 ==> -5.00
0.00, 0.00 ==> 0.00
0.00, -0.00 ==> -0.00
-0.00, 0.00 ==> 0.00
-0.00, -0.00 ==> -0.00
